src/commands/deploy/index.ts

```diff
@@ -0,0 +1,651 @@
+import { DEFAULT_VCPU, DEFAULT_MEMORY, DEFAULT_DISK_SIZE, CLOUD_URL } from "@/src/utils/constants";
+import { detectFileInCurrentDir, promptForFile } from "@/src/utils/prompts";
+import { Command, Option } from "commander";
+import dedent from "dedent";
+import fs from 'fs-extra';
+import inquirer from 'inquirer';
+import path from 'path';
+import {
+  createClient,
+  encryptEnvVars,
+  parseEnvVars,
+  ProvisionCvmComposeFileUpdateRequest,
+  safeAddComposeHash,
+  safeCommitCvmComposeFileUpdate,
+  safeCommitCvmProvision,
+  safeDeployAppAuth,
+  safeGetAppEnvEncryptPubKey,
+  safeGetAvailableNodes,
+  safeGetCvmComposeFile,
+  safeGetCvmInfo,
+  safeGetKmsList,
+  safeProvisionCvm,
+  safeProvisionCvmComposeFileUpdate,
+  type EnvVar,
+  type Client
+} from "@phala/cloud";
+import { parseDiskSizeInput, parseMemoryInput } from "@/src/utils/units";
+import { getCvmUuid, saveCvmUuid } from "@/src/utils/config";
+import { getApiKey } from '@/src/utils/credentials';
+
+interface Options {
+  name?: string;
+  compose?: string;
+  vcpu?: string;
+  memory?: string;
+  diskSize?: string;
+  image?: string;
+  nodeId?: string;
+  envFile?: string | boolean;
+  interactive?: boolean;
+  kmsId?: string;
+  uuid?: string;
+  customAppId?: string;
+  preLaunchScript?: string;
+  privateKey?: string;
+  rpcUrl?: string;
+  json?: boolean;
+  debug?: boolean;
+  apiKey?: string;
+  [key: string]: unknown;
+}
+
+async function getApiClient({ apiKey, interactive }: Readonly<Pick<Options, 'apiKey' | 'interactive'>>): Promise<Client> {
+  if (!apiKey && !process.env.PHALA_CLOUD_API_KEY) {
+    if (interactive) {
+      const { apiKey } = await inquirer.prompt([{
+        type: 'password',
+        name: 'apiKey',
+        message: 'Enter your API key:',
+        validate: (input: string) => input.trim() ? true : 'API key is required'
+      }]);
+      return createClient({ apiKey: apiKey });
+    } else {
+      const apiKey = getApiKey();
+      if (!apiKey) {
+        throw new Error(
+          'API key is required. Please provide it via --api-key or PHALA_CLOUD_API_KEY environment variable'
+        );
+      }
+      return createClient({ apiKey: apiKey });
+    }
+  } else if (apiKey) {
+    return createClient({ apiKey: apiKey });
+  }
+  return createClient();
+}
```

Comment:
It should be a share logic to other commands, so we better move it to individual source file e.g. `utils/get_api_client.ts`.

Also, all command `Options` includes `interactive` and `apiKey`, we could:

```ts
interfacce BaseCommandOptions {
  apiKey?: string;
  interactive?: boolean;
}

interface DeployOptions extends BaseCommandOptions {
  // ...
}
```